src/program/latex/diagram/wire.rs

```diff
@@ -0,0 +1,198 @@
+use std::{fmt, str::FromStr};
+
+use lazy_regex::{Lazy, Regex};
+
+use crate::{
+    expression::Expression,
+    instruction::{Gate, GateModifier},
+};
+
+use super::super::{LatexGenError, Parameter, RenderCommand, Symbol};
+
+/// The QuantikzGate is a valid cell of a QuantikzCellType that is the form of
+/// the ``Gate`` struct in the ``Instruction`` containing the components that
+/// are used to render a gate on the wire in the circuit diagram.
```

Comment:
This is a bit wordy, and probably shouldn't reference `QuantizCellType`, `Gate`, or `Instruction`. It stands on it's own as a type. Did AI write this per chance?

Reply:
It was AI assisted but I went through and reworded it pretty much entirely. If it stands on its own then I could just remove the doc.